File: HW/HW11/0320-HW11.py
def floodFill(self, image, sr, sc, color):    
    if image[sr][sc] != color:
        dfs(image, sr, sc, image[sr][sc], color)
    return image

# Define dfs function separately
def dfs(image, row, col, color, newColor):
    if row < 0 or row >= len(image) or col < 0 or col >= len(image[0]) or image[row][col] != color:
        return
    
    image[row][col] = newColor
    
    dfs(image, row + 1, col, color, newColor)
    dfs(image, row - 1, col, color, newColor)
    dfs(image, row, col + 1, color, newColor)
    dfs(image, row, col - 1, color, newColor)

File: HW/HW11/0320-HW11.java
public int[][] floodFill(int[][] image, int sr, int sc, int newColor) {
    int originalColor = image[sr][sc];
    if (originalColor != newColor) {
        dfs(image, sr, sc, originalColor, newColor);
    }//if
    return image;
}//floodfill

public void dfs(int[][] image, int row, int col, int color, int newColor) {
    if (row < 0 || row >= image.length || col < 0 || col >= image[0].length || image[row][col] != color) {
        return;
    }//if
    
    image[row][col] = newColor;
    
    dfs(image, row + 1, col, color, newColor);
    dfs(image, row - 1, col, color, newColor);
    dfs(image, row, col + 1, color, newColor);
    dfs(image, row, col - 1, color, newColor);
}//dfs